src/megaclient.cpp

```diff
@@ -12862,6 +12870,12 @@ bool MegaClient::syncdown(LocalNode* l, LocalPath& localpath, bool rubbish)
                 }
             }
         }
+
+        if (rit->second->localnode.get() == (LocalNode*)~0)
+        {
+            // make sure we are not leaving this marker pointer in the localnode
+            rit->second->localnode.store_unchecked(nullptr);
```

Comment:
I wonder if the lack of this could have cause the issue. Although looking into the code it appears it wouldn't happen for FILENODE which was the case of some reported case...
Anyway, perhaps we can have this assertion in `crossref_ptr`
```
    TO* operator->() const
   {
        assert (ptr != ~0);
        return ptr; 
}
```

Reply:
Yes, I think it's a possibility.  The assert is a good idea, I'lll add that.